Files/warranty_meraki.py

```diff
@@ -305,24 +318,34 @@ def get_all_device_serial_numbers(self, all_organization_ids, retry, timeout):
 
         return all_organizations, all_devices
 
-    # TODO: add rate limit handling
+    """
+    Status Codes
+    400: Bad Request- You did something wrong, e.g. a malformed request or missing parameter.
+    403: Forbidden- You don't have permission to do that.
+    404: Not found- No such URL, or you don't have access to the API or organization at all. 
+    429: Too Many Requests- You submitted more than 5 calls in 1 second to an Organization, triggering rate limiting. 
+    """
     def process_get_response(self, resp, retry):
+
+        self.meraki_request_throttler()
```

Comment:
Should throttling happen before the request gets made?

Reply:
The limit is 5 requests a second. the throttler keeps track of the time the last request was made and the number of requests made. if the last request made was less than 1 second ago it enters a second conditional to check if the number of requests made is divisible by 5, if it is sleep for some time to allow for more requests, finally it updates the time the last request was made and increments the number of requests made by 1.

The logic behind this was to reach that limit before going on to make the 6th API call. and since all the API calls enter this **process_get_response** function I wouldn't have to make calls to the throttler everywhere. though I haven't had the opportunity to test it yet since the calls I'm making to Meraki dashboard API are limited to the number of Meraki products I have.

Reply:
Ok makes sense.  After every 5 will be a pause so that the next request will not cause the limit to be reached.